samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -36,11 +35,11 @@ class GStreamerPipeline
         fast_measure = cmd_parser->has("fast");               // fast measure fps
         fix_fps      = cmd_parser->get<int>("fps");           // fixed frame per second
         pipeline     = cmd_parser->get<string>("pipeline"),   // gstreamer pipeline type
-        container    = cmd_parser->get<string>("container"),  // container type
         mode         = cmd_parser->get<string>("mode"),       // coding mode
         codec        = cmd_parser->get<string>("codec"),      // codec type
         file_name    = cmd_parser->get<string>("file"),       // path to videofile
         resolution   = cmd_parser->get<string>("resolution"); // video resolution
+        container    = file_name.substr(file_name.find('.') + 1);  // container type
```

Comment:
1. `find()` => `rfind()`
2. add validation for files with missing dot '.'

Reply:
@alalek done.

Reply:
@alalek replace exit_code to std::exception